pkg/codegen/nodejs/gen_test.go

```diff
@@ -28,3 +37,71 @@ func TestGenerateTypeNames(t *testing.T) {
 		}
 	})
 }
+
+// Resolves modules via Yarn and compiles the code with TypeScript
+// linking against in-repo Pulumi Node SDK.
+func compileHook() test.SdkTestHook {
+	return test.SdkTestHook{
+		Name: "compile",
+		RunHook: func(env *test.SdkTestEnv) {
+			env.Command("yarn", "install")
+			env.Command("yarn", "link", "@pulumi/pulumi")
+			tsc := env.NewCommand("tsc")
+			tsc.Env = []string{fmt.Sprintf("PATH=%s", filepath.Join(".", "node_modules", ".bin"))}
+			env.RunCommand(tsc)
+		},
+	}
+}
+
+func generatePackage(tool string, pkg *schema.Package, extraFiles map[string][]byte) (map[string][]byte, error) {
+	p := *pkg
+	if len(p.Language) > 0 {
+		panic(fmt.Sprintf("%v", p.Language))
+	}
+	if extraFiles == nil {
+		extraFiles = make(map[string][]byte)
+	}
+	nodePkgInfo := NodePackageInfo{
```

Comment:
This might need to go in schema.json if it can.

Reply:
Check out our [schema docs](https://www.pulumi.com/docs/guides/pulumi-packages/schema/#packagelanguage), which are mostly up to date. It includes `packageName` and `devDependencies` for `nodejs` projects.